storyruntime/Containers.py

```diff
@@ -139,6 +153,166 @@ def get_registry_url(cls, image):
                         f'https://{hostname}.{app.config.APP_DOMAIN}'
                         f'{expose.http_path}')
 
+    @classmethod
+    async def check_live(cls, story, line, service_name):
+        """
+        check_live is used to check if a service is live via
+         the "liveness probe"
+
+        :param story: the current story
+        :param line: the current line
+        :param service_name: the name of the service you are checking
+
+        :return: returns True if the service is live
+        """
+
+        app = story.app
+
+        container_name = cls.get_container_name(story.app, story.name,
+                                                line, service_name)
+        story.logger.debug(
+            f'Checking liveness for '
+            f'the service {service_name} with '
+            f'the container {container_name}'
+        )
+
+        res = await Kubernetes.make_k8s_call(
+            app.config, app.logger,
+            f'/apis/apps/v1/namespaces/{app.app_id}'
+            f'/deployments/{container_name}')
+
+        probe_config = Kubernetes.get_liveness_probe(story.app, service_name)
+
+        # if the container does not exist
+        # then it is in fact no live
+        if res.code != 200:
+            return False
+
+        if probe_config is None:
+            # return True since we have no idea if this
+            # pod is live or not.
+            return True
+
+        hostname = Kubernetes.get_hostname(story.app, container_name)
+
+        path = probe_config['httpGet']['path']
+        port = probe_config['httpGet']['port']
+
+        url = f'http://{hostname}:{port}{path}'
+        client = AsyncHTTPClient()
+        kwargs = {
+            'method': 'GET',
+            'connect_timeout': 5
+        }
+        exception = None
+        try:
+            result = await HttpUtils.fetch_with_retry(
+                1, story.app.logger, url, client, kwargs
```

Comment:
It's a shame we can't do:

```suggestion
                tries=1, story.app.logger, url, client, kwargs
```

Might still be better to use named arguments all around for better readability.